src/main/java/com/example/petbuddybackend/service/user/CaretakerService.java

```diff
@@ -28,4 +37,54 @@ public Page<CaretakerDTO> getCaretakers(Pageable pageable, CaretakerSearchCriter
                 .findAll(spec, pageable)
                 .map(caretakerMapper::mapToCaretakerDTO);
     }
+
+    public Page<RatingDTO> getRatings(Pageable pageable, Long caretakerId) {
+        return ratingRepository.findAllByCaretakerId(caretakerId, pageable);
+    }
+
+    public void rateCaretaker(Long caretakerId, String clientUsername, int rating, String comment) {
+        assertCaretakerExists(caretakerId);
+        Long clientId = clientService.getClientIdByUsername(clientUsername);
+
+        if(clientId.equals(caretakerId)) {
```

Comment:
tak jest szybciej ale ID trzeba pilnowac. Mozna pobierac opiekuna i klienta z bazy i patrzec czy ID konta maja takie samo, wtedy nie trzeba pilnowac tego ID, chyba ze to do jeszcze czegos by bylo

Reply:
Wyrzuciłem te id i teraz sprawdzam czy maile są takie same